Sources/Swift/Protocol/Metric.swift

```diff
@@ -0,0 +1,289 @@
+/// A metric entry that captures metric data with associated attribute metadata.
+///
+/// Use the `options.beforeSendMetric` callback to modify or filter metric data.
+public struct Metric {
+    /// A typed attribute that can be attached to structured item entries
+    public typealias Attribute = SentryAttribute
+
+    /// The timestamp when the metric was recorded.
+    public var timestamp: Date
+
+    /// The type of metric (counter, gauge, or distribution).
+    ///
+    /// - `.counter`: Incrementing integer values (e.g., request counts)
+    /// - `.gauge`: Current value at a point in time (e.g., active connections)
+    /// - `.distribution`: Statistical distribution of values (e.g., response times)
+    public let metricType: MetricType
```

Comment:
`m`: Making this let doesn't allow changing the type in hooks. If that's desired, I would add a comment explaining why you can't change it.